features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java

```diff
@@ -0,0 +1,151 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class DeviceConfigMonitor extends AbstractServiceMonitor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DeviceConfigMonitor.class);
+    private TftpServer tftpServer;
+    private SshScriptingService sshScriptingService;
+    private static final Duration DEFAULT_DURATION = Duration.ofMinutes(1); // 60sec
+    private static final long DEFAULT_TTL = 180000; //180sec
+    private static final int DEFAULT_SSH_PORT = 22;
+
+    @Override
+    public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
+        if (tftpServer == null) {
+            tftpServer = BeanUtils.getBean("daoContext", "tftpServer", TftpServer.class);
+        }
+        if (sshScriptingService == null) {
+            sshScriptingService = BeanUtils.getBean("daoContext", "sshScriptingService", SshScriptingService.class);
+        }
+        TftpReceiver tftpReceiver = new TftpReceiver(svc.getAddress());
+        try {
+            tftpServer.register(tftpReceiver);
+        } catch (IOException e) {
+            LOG.error("Exception while connecting to TFTP Server ", e);
+            return PollStatus.down("Couldn't connect to Tftp Server");
+        }
+        String script = getObjectAsStringFromParams(parameters, "script");
+        String user = getObjectAsStringFromParams(parameters, "username");
+        String password = getObjectAsStringFromParams(parameters, "password");
+        Integer portValue = getObjectAsIntFromParams(parameters, "port");
+        int port = portValue != null ? portValue : DEFAULT_SSH_PORT;
+        Long timeout = getObjectAsLongFromParams(parameters, "timeout");
+        Duration duration = timeout != null ? Duration.ofMillis(timeout) : DEFAULT_DURATION;
+        Long ttlValue = getObjectAsLongFromParams(parameters, "ttl");
+        long ttl = ttlValue != null ? ttlValue : DEFAULT_TTL;
+        Optional<SshScriptingService.Failure> sshResult =
+                sshScriptingService.execute(script, user, password, svc.getIpAddr(), port, new HashMap<>(), duration);
+        CompletableFuture<byte[]> configObj = tftpReceiver.getConfigFuture();
+        try {
+            byte[] config = configObj.get(ttl, TimeUnit.MILLISECONDS);
+            PollStatus pollStatus = PollStatus.up();
+            pollStatus.setDeviceConfig(config);
+            return pollStatus;
+        } catch (TimeoutException | InterruptedException | ExecutionException e) {
+            LOG.error("Config retrieval failed", e);
+            return PollStatus.unresponsive("Config retrieval failed : " + e.getMessage());
+        }
```

Comment:
2 things:
* the `tftpReceiver` has to be deregistered
* the `sshResult` may contain interesting failure information
```suggestion
        try {
            tftpServer.register(tftpReceiver);
            try {
                String script = getObjectAsStringFromParams(parameters, "script");
                String user = getObjectAsStringFromParams(parameters, "username");
                String password = getObjectAsStringFromParams(parameters, "password");
                Integer portValue = getObjectAsIntFromParams(parameters, "port");
                int port = portValue != null ? portValue : DEFAULT_SSH_PORT;
                Long timeout = getObjectAsLongFromParams(parameters, "timeout");
                Duration duration = timeout != null ? Duration.ofMillis(timeout) : DEFAULT_DURATION;
                Long ttlValue = getObjectAsLongFromParams(parameters, "ttl");
                long ttl = ttlValue != null ? ttlValue : DEFAULT_TTL;
                Optional<SshScriptingService.Failure> sshResult =
                        sshScriptingService.execute(script, user, password, svc.getIpAddr(), port, new HashMap<>(), duration);
                return sshResult.map(failure -> {
                    var reason = "Config retrieval could not be triggered - message: " + failure.message
                                 + "\nstdout: " + failure.stdout
                                 + "\nstderr: " + failure.stderr;
                    LOG.error(reason);
                    return PollStatus.unavailable(reason);
                }).orElseGet(()  -> {
                    CompletableFuture<byte[]> configObj = tftpReceiver.getConfigFuture();
                    try {
                        byte[] config = configObj.get(ttl, TimeUnit.MILLISECONDS);
                        PollStatus pollStatus = PollStatus.up();
                        pollStatus.setDeviceConfig(config);
                        return pollStatus;
                    } catch (TimeoutException | InterruptedException | ExecutionException e) {
                        LOG.error("Config retrieval failed", e);
                        return PollStatus.unresponsive("Config retrieval failed : " + e.getMessage());
                    }
                });
            } finally {
                tftpServer.deregister(tftpReceiver);
            }
        } catch (IOException e) {
            LOG.error("Exception while connecting to TFTP Server ", e);
            return PollStatus.down("Couldn't connect to Tftp Server");
        }
```